core/src/services/routing.js

```diff
@@ -56,22 +56,18 @@ class RoutingClass {
     if (windowPath === GenericHelpers.trimLeadingSlash(route)) {
       return;
     }
+    const hashRouting = LuigiConfig.getConfigValue('routing.useHashRouting');
+    let url = new URL(location.href);
+    hashRouting ? url.hash = route : url.pathname = route;
 
-    if (LuigiConfig.getConfigValue('routing.useHashRouting')) {
-      if (!navSync) {
-        EventListenerHelpers.hashChangeWithoutSync = true;
```

Comment:
- I'm not sure if this part needs to be removed. This part was added as part of a bug fix [here](https://github.com/SAP/luigi/pull/1893/files) for the withoutSync issue that we had before. 
- Also, I see that luigi context update when navigating internally with Angular Router, doesn't seem to work as intended. You can follow the ***'How to test'*** part of [this issues](https://github.com/SAP/luigi/pull/1953) description for the steps on reproducing it. In my environment I do not get the context change anymore in the console logs.

Reply:
We need to sync on both issues, because I think first one is resolved in the new solution and second one works for me as well.

Reply:
I double checked and it seemed like I was using an outdated luigi bundle. It seems to work well for me as well. 👍